app/controllers/admin/users_controller.rb

```diff
@@ -7,24 +7,26 @@ def index
     end
 
     def delete
-      @user.soft_delete_with_relations
+      UserCommand.new(@user).soft_delete_with_relations
       redirect_to admin_users_path(filter_by: :deleted)
     end
 
     def restore
-      @user.restore
-      @user.restore_relations
+      UserCommand.new(@user).restore_with_relations
     end
 
     def unspam
       @user.ham!
       @user.update_attribute :is_spam, false
-      @user.restore
-      @user.restore_relations
+      UserCommand.new(@user).restore_with_relations
     end
 
     def spam
-      @user.spam_and_mark_for_deletion!
```

Comment:
We should find a way to keep the "spam and delete" logic encapsulated by 1 method. There are multiple places that call `spam_and_mark_for_deletion!` and I use it on console. It was made because we don't want a visible user marked as spam in the UI.